packages/babel-core/src/transformation/util/missing-plugin-helper.js

```diff
@@ -0,0 +1,129 @@
+  },
+  exportNamespaceFrom: {
+    syntax: "@babel/plugin-syntax-export-namespace-from",
+    transform: "@babel/plugin-proposal-export-namespace-from",
+  },
+  flow: {
+    syntax: "@babel/plugin-syntax-flow",
+    transform: "@babel/plugin-transform-flow-strip-type",
+  },
+  functionBind: {
+    syntax: "@babel/plugin-syntax-function-bind",
+    transform: "@babel/plugin-proposal-function-bind",
+  },
+  functionSent: {
+    syntax: "@babel/plugin-syntax-function-sent",
+    transform: "@babel/plugin-proposal-function-sent",
+  },
+  importMeta: {
+    issueLink: "https://github.com/babel/proposals/issues/10",
+  },
+  jsx: {
+    syntax: "@babel/plugin-syntax-jsx",
+    transform: "@babel/plugin-transform-react-jsx",
+  },
+  nullishCoalescingOperator: {
+    syntax: "@babel/plugin-syntax-nullish-coalescing-operator",
+    transform: "@babel/plugin-proposal-nullish-coalescing-operator",
+  },
+  numericSeparator: {
+    syntax: "@babel/plugin-syntax-numeric-separator",
+    transform: "@babel/plugin-proposal-numeric-separator",
+  },
+  objectRestSpread: {
+    syntax: "@babel/plugin-syntax-object-rest-spread",
+    transform: "@babel/plugin-proposal-object-rest-spread",
+  },
+  optionalCatchBinding: {
+    syntax: "@babel/plugin-syntax-optional-catch-binding",
+    transform: "@babel/plugin-proposal-optional-catch-binding",
+  },
+  optionalChaining: {
+    syntax: "@babel/plugin-syntax-optional-chaining",
+    transform: "@babel/plugin-proposal-optional-chaining",
+  },
+  pipelineOperator: {
+    syntax: "@babel/plugin-syntax-pipeline-operator",
+    transform: "@babel/plugin-proposal-pipeline-operator",
+  },
+  throwExpressions: {
+    syntax: "@babel/plugin-syntax-throw-expressions",
+    transform: "@babel/plugin-proposal-throw-expressions",
+  },
+  typescript: {
+    syntax: "@babel/plugin-syntax-typescript",
+    transform: "@babel/plugin-transform-typescript",
+  },
+};
+
+export default function generateHelpMessage(
+  missingPlugins: string[] = [],
+): string {
+  let helpMessage = "";
+  const pluginName = missingPlugins.length && missingPlugins[0];
+  if (pluginName) {
+    const pluginInfo = pluginNameMap[pluginName];
+    if (pluginInfo) {
+      const {
+        syntax: syntaxPlugin,
+        transform: transformPlugin,
+        issueLink,
+      } = pluginInfo;
+      if (syntaxPlugin) {
+        if (transformPlugin) {
+          helpMessage =
+            `Use the ${syntaxPlugin} plugin to enable parsing. ` +
+            `Use the ${transformPlugin} plugin to enable transformation.` +
+            `\nNote: The transform plugins already include the syntax plugins, so you don't need both.`;
+        } else {
+          helpMessage =
+            `Use the ${syntaxPlugin} plugin to enable parsing. ` +
+            `Babel doesn't support transforming this syntax.`;
+        }
+      } else if (issueLink) {
+        helpMessage =
+          `This syntax is not supported yet. ` +
```

Comment:
`This experimental syntax`...

Reply:
And `it's` -> `its`

Reply:
It this used? That means we are adding early errors in the parser at a very early stage of the proposal, we don't do that currently?

Reply:
Will add the changes suggested in the first two comments. @xtuc I guess so. I'm not well versed with the parser yet, so I don't have a concrete answer. I just picked up all the plugins from the babylon README.